File: reservation-service/src/main/java/br/com/wellingtoncosta/reservationservice/domain/model/Reservation.java
package br.com.wellingtoncosta.reservationservice.domain.model;

import lombok.Data;

import java.time.LocalDate;

/**
 * @author Wellington Costa on 10/12/18
 */
@Data public class Reservation {

    private Long id;
    private Long bookId;
    private Long clientId;
    private LocalDate reservedAt;
    private LocalDate returnedAt;
    private String description;

}

File: reservation-service/src/main/kotlin/br/com/wellingtoncosta/reservation/Reservation.kt
package br.com.wellingtoncosta.reservation

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document
import org.springframework.format.annotation.DateTimeFormat
import org.springframework.format.annotation.DateTimeFormat.ISO.DATE_TIME
import java.util.*

/**
 * @author Wellington Costa on 25/07/18.
 */
@Document
data class Reservation(
    @Id var id: String,
    var bookId: String,
    var clientId: String,
    @DateTimeFormat(iso = DATE_TIME) var reservedAt: Date,
    @DateTimeFormat(iso = DATE_TIME) var returnedAt: Date? = null
)